src/v2/providers/pubsub.ts

```diff
@@ -3,10 +3,27 @@ import { ManifestEndpoint } from '../../runtime/manifest';
 import { CloudEvent, CloudFunction } from '../core';
 import * as options from '../options';
 
+/**
+ * A PubSub Topic is:
+ * <ul>
+ * A resource that you can publish messages to and then consume those messages via subscriptions.
+ * An isolated data stream for pubsub messages.
+ * Messages are published to a topic.
+ * Messages are listened to via a subscription.
+ * Each subscription listens to the messages published to exactly one topic.
+ */
+export type PubSubTopic = string;
+
+/**
+ * Resource that listens to the messages published by exactly one topic.
+ */
+export type PubSubSubscription = string;
+
 /**
  * Interface representing a Google Cloud Pub/Sub message.
  *
- * @param data Payload of a Pub/Sub message.
+ * @param data - Payload of a Pub/Sub message.
+ * @param T - Type representing `Message.data`'s json format
```

Comment:
Suggest "JSON" or `json` if we want the literal.

Reply:
JSON!! 🎉 🎉